### PR TITLE
Fix #52: removing `setorder` due to reproducibility-issues

### LINKED ISSUE #52: Reproducibility with parallelization
Hi, first of all thanks for that great package, which speeds up hyperparameter optimization enormously!!

I am currently stuck with getting reproducible results for xgboost on different machines using parallelization. Actually, it seems as if `set.seed()` does not have any effect. When comparing the output from `bayesOpt` appx. the first 60 rounds are identical in terms of the chosen hyperparameters and the Score, however then the results start to diverge slightly between the different machines (running the same code with the same data).

I already experimented with `parallel::clusterSetRNGStream`, however, without success so far.

The initialization of the parallel backend is as follows:

```r
cl <- parallel::makePSOCKcluster(nthread)
doParallel::registerDoParallel(cl)

on.exit(
  expr = {
    parallel::stopCluster(cl)
    foreach::registerDoSEQ()
    # reset random number generator
    RNGkind(kind = "default")
    invisible(gc())
  }
)

# export from current env
parallel::clusterExport(
  cl = cl,
  varlist = c(
    "x", "y", "params", "seed", "nthread" 
  ),
  envir = environment()
)

parallel::clusterSetRNGStream(
  cl = cl,
  iseed = seed
)
parallel::clusterEvalQ(
  cl = cl,
  expr = (
    library(xgboost)
    RNGkind("L'Ecuyer-CMRG")
    set.seed(seed) 
  }
)

set.seed(seed)
opt_obj <- ParBayesianOptimization::bayesOpt(
  ...
)
```

In addition, what is very strange that optimizing hyperparameters for `glmnet` and `ranger` using the same wrapper function to initialize the parallel backend and to parameterize `bayesOpt` indeed works and yields identical and reproducible results on both machines.

Is there anything that I might have forgot in order to get fully reproducible results when calling `bayesOpt`? (It would be really really great to have a tutorial on how to properly set up a parallel backend that produces reproducible results.)

Edit:
Could be the use of `acqThresh` with a value smaller than 1 be related to the issue / does `acqThresh` negatively affect reproducibility?

### PR DESCRIPTION
**Pull request by kapsner**
This PR fixes #52 by removing `setorder` which is currently applied to the `LocalOptims` object and leads to inconsisten results when ties occur in the sorting criteria (`gpUtility`).

### CODE REVIEW

**Comment by AnotherSamWilson** [comment]
Does this pass a local R CMD check? It looks like travis CI fails because xgboost isn't available... What a pain.

**Comment by kapsner** [comment]
I only did `devtools::test()` and it was was passed locally.

**Comment by AnotherSamWilson** [comment]
Okay I'll pull this down and run the CMD checks - if it's just Travis that's the problem we should be good to go.

**Comment by kapsner** [comment]
Let me know pls, in case of other issues I will go ahead and fix them!

**Comment by kapsner** [comment]
Hi @AnotherSamWilson , I have finally managed to verify the functioning of this branch again: running `rcmdcheck::rcmdcheck(args = "--as-cran")` resulted in the following notes:

>❯ checking for hidden files and directories ... NOTE
  Found the following hidden files and directories:
    .travis.yml
  These were most likely included in error. See section ‘Package
  structure’ in the ‘Writing R Extensions’ manual.
  
  >CRAN-pack does not know about
    .travis.yml

>❯ checking top-level files ... NOTE
  Non-standard file/directory found at top level:
    ‘cran-comments.md’

>❯ checking R code for possible problems ... NOTE
  getNextParameters: no visible binding for global variable ‘.SD’
  Undefined global functions or variables:
    .SD
  
 > Found if() conditions comparing class() to string:
  File ‘ParBayesianOptimization/R/SmallFuncs.R’: if (class(bounds) != "list") ...
  File ‘ParBayesianOptimization/R/addIterations.R’: if (class(optObj) != "bayesOpt") ...
  File ‘ParBayesianOptimization/R/bayesOpt.R’: if (class(Result) != "list") ...
  File ‘ParBayesianOptimization/R/changeSaveFile.R’: if (class(optObj) != "bayesOpt") ...
  File ‘ParBayesianOptimization/R/updateGP.R’: if (class(sgp) == "stopEarlyMsg") ...
  Use inherits() (or maybe is()) instead.

The latest commit fixes these issues. Now the results are
> 0 errors ✔ | 0 warnings ✔ | 0 notes ✔

**Comment by AnotherSamWilson** [comment]
@kapsner Sorry for the late reply, it's crunch time at work right now. Would you like to be a maintainer of this project? I'm afraid I can't give this project my full attention between work and my other open-source projects.

**Comment by kapsner** [comment]
Hello @AnotherSamWilson , thank you very much for the offer to become co-maintainer of this project. Since I also maintain several open-source projects myself and my time is also very limited, I am honestly a bit cautious with a commitment. How would you imagine such a co-maintainment?

**Comment by kapsner** [comment]
Hi @AnotherSamWilson , have you in the meantime been able to have a look at this? Can this PR be merged or is there yet missing something?
